### PR TITLE
Task job

### LINKED ISSUE #352: Retire the term "task jobs"
We define "tasks" and "jobs" as well as explaining the relationship between them.

The undefined term "task jobs" appears throughout the docs (and beyond). When I was starting out this term confused me, it's unclear how the term "job" differs from "task job". If we mean jobs we should just say jobs.

This term made a little more sense in Cylc 7 where tasks and jobs were more convoluted.

Propose removing this term, replacing it with either task or job as applicable.

### PR DESCRIPTION
**Pull request by oliver-sanders**
Closes #352

* Retire "task job" as a user-facing term in favour of the canonical
  "task" and "job" terms.
* Closes https://github.com/cylc/cylc-doc/issues/352
* The ID structure is `workflow//cycle/task/job`, they are tasks and
  jobs not "task-jobs" or "cycle-tasks" or "workflow-cycles".

Sibling: https://github.com/cylc/cylc-flow/pull/5088

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.

### CODE REVIEW

File: src/7-to-8/major-changes/platforms.rst [hunk 1]
```diff
@@ -6,46 +6,11 @@ Platforms
...
-      [runtime]
-          [[task1]]
-              [[[remote]]]
-                  host = $(rose host-select my-computer)
-          [[task2]]
-              # An example of a home-rolled host selector
-              [[[remote]]]
-                  host = $(test $((RANDOM%2)) -eq 0 && echo "host_a" || echo "host_b")
+   * :cylc:conf:`[runtime][<namespace>][job]`
+   * :cylc:conf:`[runtime][<namespace>][remote]`
```
Comment (oliver-sanders) [inline src/7-to-8/major-changes/platforms.rst:13]: Modified to keep the "Does This Change Affect Me?" short (it's meant to be quickly scan-able to help users work out whether they need to read it or not).

Full examples are provided further down (including equivalents of both of these) in the examples section.

File: src/7-to-8/major-changes/platforms.rst [hunk 2]
```diff
@@ -6,46 +6,11 @@ Platforms
...
-   Read this section if your workflow's jobs run on a remote computer or if
-   you see the following warning on running ``cylc validate``:
-
-   .. code-block:: console
-
-      WARNING - deprecated settings found (please replace with [runtime][foo]platform):
-          [runtime][foo][remote]host
-          [runtime][foo][job]batch system
-
-   If you currently use the ``rose host-select`` utility or a similar host
```
Comment (wxtim) [inline src/7-to-8/major-changes/platforms.rst:35]: I think it's worth keeping this paragraph?

File: src/user-guide/running-workflows/tracking-task-state.rst [hunk 3]
```diff
@@ -1,17 +1,17 @@
 
 .. _Task Job Polling:
 
-Task Job Polling
```
Comment (wxtim) [inline src/user-guide/running-workflows/tracking-task-state.rst:4]: Is this filename sane given the changes?

**Review by wxtim — APPROVED** [review]
- [x] Read content
- [x] Built locally
- [x] Examined built changes if change is multi-line.

I've jotted a couple of minor thoughts, neither should block this being merged.

File: src/7-to-8/major-changes/platforms.rst (hunk 2 above)
Comment (oliver-sanders) [reply src/7-to-8/major-changes/platforms.rst:35]: This is covered by `[runtime][<namespace>][remote]` (note that `rose host-select` can only work when configured by this).

**Review by datamel — APPROVED** [review]

**Comment by oliver-sanders** [comment]
(rebased)

**Comment by oliver-sanders** [comment]
Shellcheck failure is unrelated, other tests passing, tested locally.

**Pull request merged**